src/api/app/controllers/webui/patchinfo_controller.rb

```diff
@@ -193,14 +193,43 @@ def save
   end
 
   def remove
+    # checks
+    if @package.name == '_project'
+      raise DeletePackageNoPermission.new '_project package can not be deleted.'
+    end
+
+    unless User.current.can_modify_package?(@package)
+      raise DeletePackageNoPermission.new "no permission to delete package #{@package.name} in project #{@project.name}"
+    end
+
+    # deny deleting if other packages use this as develpackage
+    # Note: This might raise an exception
+    @package.can_be_deleted?
+
+    path = @package.source_path
+    parameters = {
+      user:    User.current,
+      project: @project.name,
+      package: @package.name,
+      timeout: 500
+    }
     begin
-      FrontendCompat.new.delete_package project: @project, package: @package
-      flash[:notice] = "'#{@package}' was removed successfully from project '#{@project}'"
-      Rails.cache.delete('%s_packages_mainpage' % @project)
-      Rails.cache.delete('%s_problem_packages' % @project)
-    rescue ActiveXML::Transport::Error => e
+      Package.transaction do
```

Comment:
@hennevogel For now this is just a simple replacement for using the [api source controller](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/source_controller.rb#L291). So the code is pretty similar to what is done there.
My plan was to adjust the code in a couple of follow up pull requests, including a cleanup of the current tests.

Maybe it would be better to close the PR and submit the changes all together